packages/backend/src/services/CatalogService/CatalogService.ts

```diff
@@ -495,4 +498,54 @@ export class CatalogService<
         );
         return { charts: chartAnalytics };
     }
+
+    async getMetricsCatalog(
+        user: SessionUser,
+        projectUuid: string,
+        _paginateArgs?: KnexPaginateArgs, // TODO: Pagination on `searchCatalog`
+        search?: string,
+    ): Promise<CatalogFieldWithAnalytics[]> {
+        const { organizationUuid } = await this.projectModel.getSummary(
+            projectUuid,
+        );
+
+        if (
+            user.ability.cannot(
+                'view',
+                subject('Project', { organizationUuid, projectUuid }),
+            )
+        ) {
+            throw new ForbiddenError();
+        }
+
+        const userAttributes =
+            await this.userAttributesModel.getAttributeValuesForOrgMember({
+                organizationUuid,
+                userUuid: user.userUuid,
+            });
+
+        const catalogMetrics = await this.searchCatalog(
+            projectUuid,
+            userAttributes,
+            search,
+            CatalogType.Field,
+            CatalogFilter.Metrics,
+        );
+
+        const analyticsPromises = catalogMetrics
+            .filter(
+                (item): item is CatalogField => item.type === CatalogType.Field,
```

Comment:
since we get the metrics with `CatalogFilter.Metrics` above we don't need to do anything else right?

Reply:
Yes, this is mostly for type narrowing since the  type from searchCatalog is CatalogField | CatalogTable even though we filter only metrics